### PR TITLE
(feature) Order history

### PR DESCRIPTION
**Pull request by JasonMHasperhoven**
## Prerequisites
N/A

## Task reference

https://app.asana.com/0/1125859137800433/1200141953338593

## PR description

Prepare the OrderHistory component

## Honeyframework screenshot
image

## Example app screenshot
N/A

## Storybook screenshot

image

## Checklist
  - [x] PR title has category name prefix, e.g. "\(fix\) {description}" (fix/feature/refactor/improvement/doc/test)
  - [x] Added PR description
  - [x] Relevant change in example app is verified, added example app screenshot
  - [x] Storybook: stories, docs are updated/verified
  - [ ] Tests/snapshots are updated
  - [ ] Lint check passed
  - [ ] PR development is completed, the developer is satisfied with the code quality and behavior of the app

### CODE REVIEW

**Review by bhoomij — CHANGES_REQUESTED** [review]
please add example app screenshot for orderhistory so we can make sure its not broken with this new updates. 
could you please mark done for all the checklist points, for now we have to do it manually :sweat:  until we add complete ci process

File: src/components/OrderHistory/OrderHistory.js [hunk 1]
```diff
@@ -56,31 +54,62 @@ export const OrderHistory = (props) => {
...
+    cid: PropTypes.number,
+    created: PropTypes.number,
+    gid: PropTypes.number,
+    id: PropTypes.number,
+    originalAmount: PropTypes.number,
+    price: PropTypes.number,
+    status: PropTypes.string,
+    symbol: PropTypes.string,
+    type: PropTypes.string,
+  })),
```
[the remark this answers is not in the document]
Comment (JasonMHasperhoven) [reply src/components/OrderHistory/OrderHistory.js:81]: Understood

File: src/components/ui/Table/_Table.scss [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
-$table-border-width: 1px !default;
-$table-border-color: $ufx-divider-black !default;
-$table-header-font-weight: $ufx-font-weight-bold;
+$ufx-table-row-height: $ufx-base-size * 4 !default;
+$ufx-table-row-height-small: $ufx-base-size * 3 !default;
+$ufx-table-border-width: 1px !default;
+$ufx-table-border-color: $ufx-divider-black !default;
+$ufx-table-header-font-weight: $ufx-font-weight-bold;
+
+$ufx-table-even-row-background-color: white !default;
```
[the remark this answers is not in the document]
Comment (JasonMHasperhoven) [reply src/components/ui/Table/_Table.scss:7]: I wasn't aware of this, good suggestion

File: src/containers/OrderHistory/OrderHistory.constants.js [hunk 3]
```diff
@@ -2,17 +2,19 @@
 import _get from 'lodash/get'
 import React from 'react'
 
-import { ORDER_HISTORY_KEYS as KEYS } from '../../components'
+import { ORDER_HISTORY_COLUMNS } from '../../components'
```
[the remark this answers is not in the document]
Comment (JasonMHasperhoven) [reply src/containers/OrderHistory/OrderHistory.constants.js:5]: Yes will do it in a separate PR, or revert to KEYS as it's no longer needed.

[the remark this answers is not in the document]
Comment (bhoomij) [reply src/containers/OrderHistory/OrderHistory.constants.js:5]: yes, good if we revert to KEYS in this PR, and then in separate PR update at all relevant places

File: src/containers/OrderHistory/OrderHistory.container.js [hunk 4]
```diff
@@ -18,4 +19,4 @@ const OrderHistoryContainer = (props) => {
   )
 }
 
-export default memo(OrderHistoryContainer)
+export default withResponsive(memo(OrderHistoryContainer))
```
Comment (bhoomij) [inline src/containers/OrderHistory/OrderHistory.container.js:22]: I think withResponsive is not needed now in container

File: src/components/OrderHistory/stories/OrderHistory.stories.js [hunk 5]
```diff
@@ -13,6 +14,18 @@ const props = {
...
+export const customColumns = showTemplateStory(Component, {
+  ...props,
+  columns: [
+    ORDER_HISTORY_COLUMNS.PRICE,
+    ORDER_HISTORY_COLUMNS.AMOUNT,
+    ORDER_HISTORY_COLUMNS.TYPE,
+    ORDER_HISTORY_COLUMNS.STATUS,
+    ORDER_HISTORY_COLUMNS.PRICE_AVERAGE,
+    ORDER_HISTORY_COLUMNS.PAIR,
+  ],
```
Comment (bhoomij) [inline src/components/OrderHistory/stories/OrderHistory.stories.js:26]: columns should be removed as we removed that prop ?

File: src/components/OrderHistory/OrderHistory.Row.js [hunk 6]
```diff
@@ -16,17 +17,20 @@ const OrderHRow = (props) => {
   } = props
   const { t } = useTranslation('orderhistory')
 
+  const visibleColumns = getVisibleColumns(columns, customMapping)
+  const orderedColumns = getOrderedColumns(visibleColumns, customMapping)
```
Comment (bhoomij) [inline src/components/OrderHistory/OrderHistory.Row.js:21]: should we memoize columns ?

**Review by bhoomij — COMMENTED** [review]
@JasonMHasperhoven , if this is ready now, could you please complete following so we can proceed to merge it ? Thanks!
- revert back ORDER_HISTORY_COLUMNS to ORDER_HISTORY_KEYS
- add example app screenshot
- mark remaining checklist points as done

**Review by jeojoe — APPROVED** [review]

**Pull request merged**